src/main/java/com/google/sps/servlets/GetGameDialogue.java

```diff
@@ -24,21 +29,24 @@
 /** Servlet that should respond with a JSON String that contains the desired dialogue */
 @WebServlet("/game-dialogue")
 public final class GetGameDialogue extends HttpServlet {
-  private static final String TEXT_TO_HTML = "text/html;";
   private static final String JSON_CONTENT_TYPE = "application/json";
-  private static final String SOFTWAREID = "software-engineering-0";
-  private static final String TESTCONTENT = "hello this is a test";
-
-  // TODO: GameStage will change and be query from a database that will be implemented
-  private GameStage currentGameStage = new GameStage(SOFTWAREID, TESTCONTENT);
+  private static final Gson gson = new Gson();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private final PlayerDatabase playerDatabase = new PlayerDatabase(datastore);
+  private final GameStageDatabase gameStageDatabase = new GameStageDatabase(datastore);
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      // you can only use the PlayerDatabase sucessfully if you are running this on a live server
+      // because it requeres you to log in and there is no way to mock a user
+      GameStage currentGameStage =
+          gameStageDatabase.getGameStage(playerDatabase.getEntityCurrentPageID());
+      String dialogue = gson.toJson(currentGameStage.getContent());
 
-    Gson gson = new Gson();
-    String dialogue = gson.toJson(currentGameStage.getContent());
-
-    response.setContentType(JSON_CONTENT_TYPE);
-    response.getWriter().println(dialogue);
+      response.setContentType(JSON_CONTENT_TYPE);
+      response.getWriter().println(dialogue);
+    } catch (Exception nullPointer) {
```

Comment:
Why do we ignore the exception? Can we catch a more specific exception?

Reply:
+1